llpc/patch/llpcPatchBufferOp.cpp

```diff
@@ -1521,31 +1520,63 @@ bool PatchBufferOp::RemoveUsersForInvariantStarts(
 }
 
 // =====================================================================================================================
-// Replace a fat pointer load with the intrinsics required to do the load.
-Value* PatchBufferOp::ReplaceLoad(
-    LoadInst* const pLoadInst) // [in] The load instruction to replace.
+// Replace a fat pointer load or store with the required intrinsics.
+Value* PatchBufferOp::ReplaceLoadStore(
+    LoadInst* const pLoadInst, // [in] The load instruction to replace.
+    StoreInst* const pStoreInst) // [in] The store instruction to replace.
 {
-    m_pBuilder->SetInsertPoint(pLoadInst);
-
-    Value* const pPointer = GetPointerOperandAsInst(pLoadInst->getPointerOperand());
+    // Either load instruction or store instruction is valid (not both)
+    LLPC_ASSERT((pLoadInst == nullptr) != (pStoreInst == nullptr));
+
+    bool isLoad = pLoadInst != nullptr;
+    Instruction* pInst = nullptr;
+    Type* pType = nullptr;
+    Value* pPointerOperand = nullptr;
+    AtomicOrdering ordering = AtomicOrdering::NotAtomic;
+    uint32_t alignment = 0;
+    SyncScope::ID syncScopeID = 0;
+
+    if (isLoad)
+    {
+        pInst = pLoadInst;
+        pType = pLoadInst->getType();
+        pPointerOperand = pLoadInst->getPointerOperand();
+        ordering = pLoadInst->getOrdering();
+        alignment = pLoadInst->getAlignment();
+        syncScopeID = pLoadInst->getSyncScopeID();
+    }
+    else
+    {
+        pInst = pStoreInst;
+        pType = pStoreInst->getValueOperand()->getType();
+        pPointerOperand = pStoreInst->getPointerOperand();
+        ordering = pStoreInst->getOrdering();
+        alignment = pStoreInst->getAlignment();
+        syncScopeID = pStoreInst->getSyncScopeID();
+    }
```

Comment:
Please instead just pass pInst as an Instruction* to the function, and use dyn_cast here. Note that dyn_cast (and cast, and isa) is very fast as it does not use RTTI.